unblob/handlers/archive/cpio.py

```diff
@@ -92,37 +179,78 @@ def calculate_chunk(  # noqa: C901
-            current_offset += self._pad_content(header, c_filesize, c_namesize)
-
-        end_offset = start_offset + self._pad_file(
-            file_with_offset, current_offset - start_offset
-        )
-        if start_offset == end_offset:
-            return None
-        return ValidChunk(
-            start_offset=start_offset,
-            end_offset=end_offset,
-        )
-
-    @classmethod
-    def _pad_file(cls, file: OffsetFile, end_offset: int) -> int:
+                raise InvalidInputFormat("CPIO entry mode is invalid.")
+
+            if self.valid_checksum(header, current_offset):
+                self.entries.append(
+                    CPIOEntry(
+                        start_offset=current_offset,
+                        size=c_filesize,
+                        dev=self._calculate_dev(header),
+                        mode=c_mode,
+                        rdev=self._calculate_rdev(header),
+                        path=Path(filename),
+                    )
+                )
+            else:
+                logger.warning("Invalid CRC for CPIO entry, skipping.", header=header)
+
+            current_offset += self._pad_content(c_filesize)
+
+        self.end_offset = self._pad_file(current_offset)
+        if self.start_offset == self.end_offset:
+            raise InvalidInputFormat("Invalid CPIO archive.")
+
+    def dump_entries(self, fs: FileSystem):
+        for entry in self.entries:
+            # skip entries with "." as filename
+            if entry.path.name in ("", "."):
+                continue
+
+            if stat.S_ISREG(entry.mode):
+                fs.carve(entry.path, self.file, entry.start_offset, entry.size)
+            elif stat.S_ISDIR(entry.mode):
+                fs.mkdir(
+                    entry.path, mode=entry.mode & 0o777, parents=True, exist_ok=True
+                )
+            elif stat.S_ISLNK(entry.mode):
+                link_path = Path(
+                    snull(
+                        self.file[entry.start_offset : entry.start_offset + entry.size]
+                    ).decode("utf-8")
+                )
+                fs.create_symlink(src=link_path, dst=entry.path)
+            elif (
+                stat.S_ISCHR(entry.mode)
+                or stat.S_ISBLK(entry.mode)
+                or stat.S_ISSOCK(entry.mode)
+                or stat.S_ISSOCK(entry.mode)
+            ):
+                fs.mknod(entry.path, mode=entry.mode & 0o777, device=entry.rdev)
+            else:
+                logger.warning("unknown file type in CPIO archive")
+
+    def _pad_file(self, end_offset: int) -> int:
         """CPIO archives can have a 512 bytes block padding at the end."""
-        file.seek(end_offset, io.SEEK_SET)
-        padded_end_offset = round_up(end_offset, cls._FILE_PAD_ALIGN)
+        self.file.seek(end_offset, io.SEEK_SET)
+        padded_end_offset = self.start_offset + round_up(
+            end_offset, self._FILE_PAD_ALIGN
```

Comment:
```suggestion
            size=end_offset - self.start_offset,
            alignment=self._FILE_PAD_ALIGN,
```

Reply:
Applied your suggestion. Thanks for you help during this review !